lib/leanplum_api/api.rb

```diff
@@ -266,5 +299,26 @@ def build_event_attributes_hash(event_hash, options = {})
 
       event_hash.keys.size > 0 ? event.merge(params: event_hash.symbolize_keys ) : event
     end
+
+    def build_send_message(message_id:, user_id:, device_id: nil, create_disposition: "CreateNever", force: false, values: {}, dev_mode: false)
+      {
+        action: 'sendMessage',
+        messageId: message_id,
+        userId: user_id,
+        deviceId: device_id,
+        createDisposition: create_disposition,
+        force: force,
+        values: values.to_json,
+        devMode: dev_mode
+      }
+    end
+
+    def users_valid?(users)
```

Comment:
This is interesting logic. Lets write a test for this or at least add a comment for this method describing the error messaging paradigm.

Reply:
Tests exist for this. I've re-worked how this works and will push an update.